Refactor/event calendar

kulish-alina opened the pull request:
#295
#309

kulish-alina committed:
Feature:
- periodical month table
- year selector/changer
- hover detailed information about dayly event
- highlinting with different colors user's events

Refactor:
- css
- style

Fix:
- Datepicker: input instead of slider; date-time conversion;

kulish-alina committed:
small fix in VacancyService

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+         <div ng-bind="day" ng-repeat="day in daysOfWeek" class="div-table-col-head" align="center"></div>
+      </div>
+      <div class="div-table-row">
+         <div ng-click="getEventsForDate(day)" ng-dblclick="onDblclick(day)" ng-class="{'today-date': (todayDate | date:'dd, MM, yyyy') === (day | date:'dd, MM, yyyy'), 'dates-from-previous-month': (dateForSearchEvents | date:'MM') !== (day | date:'MM')}" class="div-table-col" ng-repeat="day in daysInMonth | limitTo : 7 : 0">
```
Extract ng-class expression

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               <div class="small-10 grid-block">
+                  <p class="day-title" ng-bind="day | date:'d'"></p>
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
```
extract ng-if expression into a method or something

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
+                  <p class="events-number">+</p>
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
```
extract ng-bind expression into a method or something

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
+               </div>
+            </div>
+            <events user-color="userColor" selected-date="day" get-events-by-date="getEventsByDate" user-id="1" vacancies="vacancies" candidates="candidates" event-types="eventTypes" responsibles="users" remove="removeCurrentEvent" save="saveCurrentEvent" events="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()]" source="calendar" type="only-events"></events>
```
extract events parameter expression into a field or method

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+         </div>
+      </div>
+      <div class="div-table-row">
+         <div ng-click="getEventsForDate(day)" ng-dblclick="onDblclick(day)" ng-class="{'today-date': (todayDate | date:'dd, MM, yyyy') === (day | date:'dd, MM, yyyy'), 'dates-from-previous-month': (dateForSearchEvents | date:'MM') !== (day | date:'MM')}" class="div-table-col" ng-repeat="day in daysInMonth | limitTo : 7 : 7">
```
extract ng-class expression into a field or method

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               <div class="small-10 grid-block">
+                  <p class="day-title" ng-bind="day | date:'d'"></p>
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
```
extract ng-if expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
+               </div>
+            </div>
+            <events user-color="userColor" selected-date="day" get-events-by-date="getEventsByDate" user-id="1" vacancies="vacancies" candidates="candidates" event-types="eventTypes" responsibles="users" remove="removeCurrentEvent" save="saveCurrentEvent" events="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()]" source="calendar" type="only-events"></events>
```
extract event parameter expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+         </div>
+      </div>
+      <div class="div-table-row">
+         <div ng-click="getEventsForDate(day)" ng-dblclick="onDblclick(day)" ng-class="{'today-date': (todayDate | date:'dd, MM, yyyy') === (day | date:'dd, MM, yyyy'), 'dates-from-previous-month': (dateForSearchEvents | date:'MM') !== (day | date:'MM')}" class="div-table-col" ng-repeat="day in daysInMonth | limitTo : 7 : 14">
```
extract ng-class expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               <div class="small-10 grid-block">
+                  <p class="day-title" ng-bind="day | date:'d'"></p>
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
```
extract ng-if expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
+               </div>
+            </div>
+            <events user-color="userColor" selected-date="day" get-events-by-date="getEventsByDate" user-id="1" vacancies="vacancies" candidates="candidates" event-types="eventTypes" responsibles="users" remove="removeCurrentEvent" save="saveCurrentEvent" events="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()]" source="calendar" type="only-events"></events>
```
extract events parameter into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+         </div>
+      </div>
+      <div class="div-table-row">
+         <div ng-click="getEventsForDate(day)" ng-dblclick="onDblclick(day)" ng-class="{'today-date': (todayDate | date:'dd, MM, yyyy') === (day | date:'dd, MM, yyyy'), 'dates-from-previous-month': (dateForSearchEvents | date:'MM') !== (day | date:'MM')}" class="div-table-col" ng-repeat="day in daysInMonth | limitTo : 7 : 21">
```
extract ng-class expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               <div class="small-10 grid-block">
+                  <p class="day-title" ng-bind="day | date:'d'"></p>
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
```
extract ng-if expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
+               </div>
+            </div>
+            <events user-color="userColor" selected-date="day" get-events-by-date="getEventsByDate" user-id="1" vacancies="vacancies" candidates="candidates" event-types="eventTypes" responsibles="users" remove="removeCurrentEvent" save="saveCurrentEvent" events="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()]" source="calendar" type="only-events"></events>
```
extract events parameter into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+         </div>
+      </div>
+      <div class="div-table-row">
+         <div ng-click="getEventsForDate(day)" ng-dblclick="onDblclick(day)" ng-class="{'today-date': (todayDate | date:'dd, MM, yyyy') === (day | date:'dd, MM, yyyy'), 'dates-from-previous-month': (dateForSearchEvents | date:'MM') !== (day | date:'MM')}" class="div-table-col" ng-repeat="day in daysInMonth | limitTo : 7 : 28">
```
extract ng-class expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               <div class="small-10 grid-block">
+                  <p class="day-title" ng-bind="day | date:'d'"></p>
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
```
extract ng-if expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
+               </div>
+            </div>
+            <events user-color="userColor" selected-date="day" get-events-by-date="getEventsByDate" user-id="1" vacancies="vacancies" candidates="candidates" event-types="eventTypes" responsibles="users" remove="removeCurrentEvent" save="saveCurrentEvent" events="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()]" source="calendar" type="only-events"></events>
```
extract events parameter into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+         </div>
+      </div>
+      <div class="div-table-row">
+         <div ng-click="getEventsForDate(day)" ng-dblclick="onDblclick(day)" ng-class="{'today-date': (todayDate | date:'dd, MM, yyyy') === (day | date:'dd, MM, yyyy'), 'dates-from-previous-month': (dateForSearchEvents | date:'MM') !== (day | date:'MM')}" class="div-table-col" ng-repeat="day in daysInMonth | limitTo : 7 : 35">
```
extract ng-class expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               <div class="small-10 grid-block">
+                  <p class="day-title" ng-bind="day | date:'d'"></p>
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
```
extract ng-if expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
+                  <p class="events-number">+</p>
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
```
extract ng-bind expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
+               </div>
+            </div>
+            <events user-color="userColor" selected-date="day" get-events-by-date="getEventsByDate" user-id="1" vacancies="vacancies" candidates="candidates" event-types="eventTypes" responsibles="users" remove="removeCurrentEvent" save="saveCurrentEvent" events="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()]" source="calendar" type="only-events"></events>
```
extract events parameter into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
+                  <p class="events-number">+</p>
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
```
extract ng-bind expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
+                  <p class="events-number">+</p>
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
```
extract ng-bind expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
+                  <p class="events-number">+</p>
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
```
extract ng-bind expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.directive.html:
```diff
@@ -1,44 +1,108 @@
+               </div>
+               <div class="small-2 grid-block" ng-if="(eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3) > 0">
+                  <p class="events-number">+</p>
+                  <p class="events-number" ng-bind="eventsForMonth[day.getDate()+','+(day.getMonth()+1)+','+day.getFullYear()].length-3"></p>
```
extract ng-bind expression into method or field

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.js:
```diff
@@ -33,59 +35,82 @@ function EventCalendarController($scope) {
+   (function init() {
+      formatingSearchCondition();
+      getDaysInMonth(vm.startDate, vm.endDate);
+      vm.getEvents(vm.startDate, vm.endDate, vm.checkedUsers).then((events) => {
```
you can call method in 'then' instead of creating a new clause. like 
**vm.getEvents(...).then(convertEventsToHash);**

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.js:
```diff
@@ -95,45 +120,71 @@ function EventCalendarController($scope) {
-      vm.getEvents(new Date(vm.currentYear, vm.currentMonth, 1), vm.checkedUsers).then((events) => {
+      formatingSearchCondition();
+      getDaysInMonth(vm.startDate, vm.endDate);
+      vm.getEvents(vm.startDate, vm.endDate, vm.checkedUsers).then((events) => {
```
you can call method in 'then' instead of creating a new clause. like 
**vm.getEvents(...).then(convertEventsToHash);**

mykytayefrueiv commented on src/wwwroot/app/services/EventsService.js:
```diff
@@ -70,8 +70,9 @@ export default class EventsService {
       event.eventDate = utils.formatDateTimeFromServer(event.eventDate);
-      this._fillEntities(event);
-      this._convertIdsToString(event);
+      this._fillEntities(event).then(() => {
```
you can call method in 'then' instead of creating a new clause. like 
**this._fillEntities(...).then(this._convertIdsToString(event));**

mykytayefrueiv commented on src/wwwroot/app/services/NoteService.js:
```diff
@@ -12,12 +16,17 @@ export default class NoteService {
+      this.convertDateToServerFormat(entity);
       if (entity.id) {
-         return _HttpService.put(`${NOTE_URL}/${entity.id}`, entity);
+         return _HttpService.put(`${NOTE_URL}/${entity.id}`, entity).then((comment) => {
```
you can call method in 'then' instead of creating a new clause. like 
**_HttpService.put(`...).then(this.convertDateFromServerFormat)**

mykytayefrueiv commented on src/wwwroot/app/services/NoteService.js:
```diff
@@ -12,12 +16,17 @@ export default class NoteService {
          let currentUser = _UserService.getCurrentUser();
          entity.userId = currentUser.id;
-         return _HttpService.post(NOTE_URL, entity);
+         return _HttpService.post(NOTE_URL, entity).then((comment) => {
```
you can call method in 'then' instead of creating a new clause. like 
**_HttpService.post(...).then(this.convertDateFromServerFormat)**

mykytayefrueiv commented on src/wwwroot/app/views/calendar/calendar.controller.js:
```diff
@@ -32,21 +42,50 @@ export default function CandidateProfileController(
-      UserService.getUsers().then(users => set(vm, 'users', users));
+      UserService.getUsers().then(users => {
+         set(vm, 'users', users);
+         genereteColorsForUsers();
```
generate , not generete

kulish-alina committed:
ng-bind and ng-if expressions were extracted into method

kulish-alina committed:
fix bug with saving comments

kulish-alina committed:
add button back in vacancy.edit.controller and candidate.controller.js

kulish-alina committed:
small bug fix with file uploader in candidate.profile.html

kulish-alina committed:
Merge branch 'development' of https://github.com/kulish-alina/HR_Project into refactor/event-calendar

kulish-alina committed:
convert comment from candidate in CandidateService

kulish-alina committed:
buttons back in candidate and vacancy;

back to previous page after saving;
fix bug with mapping in comment

kulish-alina committed:
fix bugs with state go

kulish-alina committed:
Merge branch 'development' of https://github.com/kulish-alina/HR_Project into refactor/event-calendar

Conflicts:
	src/BaseOfTalents/DAL/DAL.csproj
	src/BaseOfTalents/DAL/Migrations/201607251323483_init.Designer.cs
	src/BaseOfTalents/DAL/Migrations/201607251323483_init.cs
	src/BaseOfTalents/DAL/Migrations/201608080846148_init.Designer.cs
	src/BaseOfTalents/DAL/Migrations/201608080846148_init.cs
	src/BaseOfTalents/DAL/Migrations/201608091326206_init.Designer.cs
	src/BaseOfTalents/DAL/Migrations/201608091326206_init.cs
	src/BaseOfTalents/DAL/Migrations/DummyData.cs
	src/wwwroot/app/main.scss
	src/wwwroot/app/services/VacancyService.js
	src/wwwroot/app/views/candidate.profile/candidate.profile.html
	src/wwwroot/app/views/candidate/candidate.view.html
	src/wwwroot/app/views/vacancy.profile/vacancy.view.html

kulish-alina committed:
user events without reloading

kulish-alina committed:
fix bug in event-calendar

kulish-alina committed:
bug with view comments in vacancy was fixed

kulish-alina committed:
fix bugs with state

kulish-alina committed:
Merge branch 'development' of https://github.com/kulish-alina/HR_Project into refactor/event-calendar

kulish-alina committed:
fix bug with attach candidates to vacancy

kulish-alina committed:
ability to turn on or off candidate's age in a search parameters candidates was implemented;

buttons 'cancel' in vacancies and candidates were renamed to 'clear', this buttons were replaced to search sidebar;

horba commented on src/wwwroot/app/directives/event-calendar/event-calendar.js:
```diff
@@ -29,63 +33,106 @@ export default class EventCalendarDirective {
    });
 
-   function getDaysInMonth(month, year) {
+   function getDaysInMonth(startDt, endDt) {
```
why are these arg's names is better?

horba replied:
this method return nothing as I see, so it can't be get-method. As well as your method not only set days in month, but also set current-date and execute other functions.  So you can remake this method into get-function or rename it.

horba commented on src/wwwroot/app/directives/event-calendar/event-calendar.js:
```diff
@@ -29,63 +33,106 @@ export default class EventCalendarDirective {
-      while (date.getMonth() === month) {
-         vm.daysInMonth.push(new Date(date));
-         date.setDate(date.getDate() + 1);
+      vm.currentDate = new Date(startDt);
```
what difference between vm.currentDate and todayDate? what function each variable?

horba commented on src/wwwroot/app/directives/event-calendar/event-calendar.js:
```diff
@@ -29,63 +33,106 @@ export default class EventCalendarDirective {
+   vm.getDailyEvents            = getDailyEvents;
+   vm.numberEventsForDailyView  = 3;
+   vm.lengthOfWeek              = 7;
+   vm.daysOfWeek                = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
```
may be is 'vm.daysOfWeekNames' better?

horba commented on src/wwwroot/app/services/NoteService.js:
```diff
@@ -32,6 +41,19 @@ export default class NoteService {
       let currentUser = _UserService.getCurrentUser();
-      return _HttpService.get(`${NOTE_URL}/user/${currentUser.id}`);
+      return _HttpService.get(`${NOTE_URL}/user/${currentUser.id}`).then((comments) => {
+         return each(comments, (comment) => {
```
it can be:
_HttpService.get(`${NOTE_URL}/user/${currentUser.id}`).then(partialRight(each, this.convertDateFromServerFormat));

horba commented on src/wwwroot/app/services/VacancyService.js:
```diff
@@ -74,6 +75,7 @@ export default class VacancyService {
+         vacancy.comments = promises[3];
          assignIn(vacancy, promises[1]);
          vacancy.childVacancies = promises[2];
          vacancy.closingCandidate = promises[3];
```
can you name indexes in promises array?

horba commented on src/wwwroot/app/services/VacancyService.js:
```diff
@@ -134,14 +137,36 @@ export default class VacancyService {
+      }
+   }
+
+   _convertCommentsToServer(vacancy) {
```
should this function return something?

kulish-alina replied:
No, this method should convert date and remove property 'responsible' from comment

kulish-alina committed:
bug with the creation of a new candidate was fixed;

bug with updating model in candidate was fixed;

gasty commented on src/wwwroot/app/services/VacancyService.js:
```diff
@@ -134,14 +137,36 @@ export default class VacancyService {
+      }
+   }
+
    _getChildVacancies(vacancy) {
```
get-function have to return some value

mykytayefrueiv commented on src/wwwroot/app/directives/event-calendar/event-calendar.js:
```diff
@@ -29,63 +33,106 @@ export default class EventCalendarDirective {
+         });
+      }
+      vm.checkedUsers = clonedCheckedUsersIds;
    });
```
extract if-else clauses into a methods. (As for a wish you can extract conditions in **if** too)

kulish-alina committed:
corrections to pull request:

 - indexes in promises array were named;
 - renamed method getDaysInMonth() to formingDateArrayForCalendar() and getEventsForDate() to selectionEventsForSpecificDay();
 - rename vm.daysOfWeek to vm.namesOfWeekdays
